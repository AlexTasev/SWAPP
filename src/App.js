import React, { useState } from 'react';
import './App.css';
import ApolloClient from "apollo-client";
import { ApolloProvider } from "react-apollo";
import { BrowserRouter } from 'react-router-dom';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import Home from './Home';
import authLink from './client/auth';
import {typeDefs} from './client/local'
import { ThemeProvider } from 'styled-components';
import { lightTheme, darkTheme } from './theme';
import { GlobalStyles } from './global';

const cache = new InMemoryCache();

const httpLink = createHttpLink({
  uri: 'http://softuni-swapp-212366186.eu-west-1.elb.amazonaws.com/graphql',
});

const client = new ApolloClient({
  cache,
  link: authLink.concat(httpLink),
  typeDefs
});

const isAuthed = !!localStorage.getItem('token');


cache.writeData({
  data: {
    authenticated: isAuthed,
  },
});


function App() {

  const [theme, setTheme] = useState('light');
  const toggleTheme = () => {
    if (theme === 'light') {
      setTheme('dark');
    } else {
      setTheme('light');
    }
  }

  console.log(toggleTheme);
  
  return (
    <ApolloProvider client={client}>
      <BrowserRouter>
      <div className="App">
      <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
      <GlobalStyles />
        <Home toggleTheme = {{toggleTheme}}/>
      </ThemeProvider>
      </div>
      </BrowserRouter>
    </ApolloProvider>
  );
}

export default App;