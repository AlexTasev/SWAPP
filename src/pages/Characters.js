import React, {Fragment} from 'react';
import { Query } from "react-apollo";
import { Link as RouterLink } from 'react-router-dom';
import { Box, Heading, Link, Button } from 'rebass';
import { useQuery, useMutation } from '@apollo/react-hooks';

import gql from "graphql-tag";
import RedirectToLogin from '../components/RedirectToLogin';

const charactersQuery = gql`
  query CharactersQuery($first: Int!, $after: String) {
    allPeople(first: $first, after: $after){
      edges{
        node{
          id
          image
          name
        }
      }
      pageInfo{
        hasNextPage
        endCursor
      }
    }
      
    }
`;

const Characters = () => {
  let first = 12;

const {data, loading, error, fetchMore} = useQuery(charactersQuery, {
  variables: {first}
})

if (loading) return <p>Loading...</p>;
if (error)return (
  <RedirectToLogin/>
)

const [...allPeople] = data.allPeople.edges;
let {hasNextPage, endCursor} = data.allPeople.pageInfo;


const loadMoreResults = () => {
  fetchMore({
    variables: {
      after: endCursor,
      first: 12
    },
    updateQuery: (prev, { fetchMoreResult: {allPeople}}) => {
              
              return {
                allPeople: {
                  ...allPeople,
                    edges: [
                    ...prev.allPeople.edges, 
                    ...allPeople.edges]
                },
              };
    },
  });
};

const characterLinsting= allPeople.map(({ node }) => (
  <div key={node.id} className="content">
      <Link  className="blog-entry"
          as={RouterLink}
          variant="nav"       
          key={node.id}
          to={`/characters/${node.id}`}>
        <img src={node.image} className="attachment-post-thumbnail size-post-thumbnail wp-post-image" alt='starwars-episode-thumbnail'/>
        <div className="blog-content-body">
          <h2 className="header">{node.name}</h2>
        </div>
      </Link>
  </div>
));

return (
  <Fragment>
    {characterLinsting}
    {hasNextPage && (
        <Box m={3} textAlign="center">
          <Button onClick={loadMoreResults}>Load More</Button>
        </Box>
      )}
  </Fragment>
);

}

export default Characters;