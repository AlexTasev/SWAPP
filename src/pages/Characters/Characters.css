.page-content {
    display: flex;
    flex-wrap: wrap;
    width: 90%;
    padding-top: 2em;
    margin: 0 auto;
}

.character-box{
    width: 30%;
    margin: 15px 0;
    display: block;
    margin: 0 auto;
    padding: 5px;
}



.character-content-body {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

@media (max-width:791px){
    .character-box {
        width: 100%;
    }
}