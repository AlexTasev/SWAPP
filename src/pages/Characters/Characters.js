import React, {Fragment} from 'react';
import { NavLink } from 'react-router-dom';
import { useQuery } from '@apollo/react-hooks';
import './Characters.css'
import gql from "graphql-tag";
import RedirectToLogin from '../../components/RedirectToLogin';
import Loading from '../../components/LoginForm/Loading';

const charactersQuery = gql`
  query CharactersQuery($first: Int!, $after: String) {
    allPeople(first: $first, after: $after){
      edges{
        node{
          id
          image
          name
        }
      }
      pageInfo{
        hasNextPage
        endCursor
      }
    }
      
    }
`;

const Characters = () => {
  let first = 12;

  const { data, loading, error, fetchMore } = useQuery(charactersQuery, {
    variables: { first }
  })

  if (loading) return <Loading/>;
  if (error)return <RedirectToLogin/>

  const [...allPeople] = data.allPeople.edges;
  let { hasNextPage, endCursor } = data.allPeople.pageInfo;


  const loadMoreResults = () => {
    fetchMore({
      variables: {
        after: endCursor,
        first: 12
      },
      updateQuery: (prev, { fetchMoreResult: { allPeople } }) => {

        return {
          allPeople: {
            ...allPeople,
            edges: [
              ...prev.allPeople.edges,
              ...allPeople.edges]
          },
        };
      },
    });
  };

  const characterLinsting = allPeople.map(({ node }) => (
    <div key={node.id} className="character-box">
      <NavLink className="character-entry"
        variant="nav"
        key={node.id}
        to={`/characters/${node.id}`}>
        <img src={node.image} alt='starwars-episode-thumbnail' />
        <div className="character-content-body">
          <h2 className="header">{node.name}</h2>
        </div>
      </NavLink>
    </div>
  ));

  return (
    <Fragment>
    <div className="page-content">
      {characterLinsting}
    </div>
    {hasNextPage && (
      <div className="page-content center-text">
        <button onClick={loadMoreResults}>Load More</button>
      </div>
    )}
  </Fragment>
  );

}

export default Characters;