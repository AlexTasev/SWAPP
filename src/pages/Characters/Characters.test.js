import React from 'react';
import Characters from './Characters'
import { ApolloProvider } from "react-apollo";
import {client} from '../../App'
const ReactTestRenderer = require('react-test-renderer');

describe('My second snapshot test', () => {
    it('Should compare the component with a snapshot', () => {
      const component = ReactTestRenderer.create(
        <ApolloProvider client={client}>
          <Characters />
        </ApolloProvider>
      );
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    })
  });