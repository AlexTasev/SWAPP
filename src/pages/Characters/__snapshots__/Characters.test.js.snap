// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My second snapshot test Should compare the component with a snapshot 1`] = `
<p
  className="loading"
>
  Loading...
</p>
`;
